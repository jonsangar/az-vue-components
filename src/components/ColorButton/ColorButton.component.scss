.colorButton{

  &__fill{
    position: relative;
    display: inline-block;
    width: 30px;
    height: 30px;
    border-radius: 50%;


    &__color-1{
      background-image: radial-gradient(circle at 100% 100%,#52606D 0%, #9AA5B1 100%);

      &--selected{
        &::after{
          position: absolute;
          top: -4px;
          right: 0;
          bottom: 0;
          left: -4px;
          width: 38px;
          height: 38px;
          content: '';
          border: 2px solid #52606D;
          border-radius: 50%;
        }
      }
    }

    &__color-2{
      background-image: radial-gradient(circle at 100% 100%,$color-red-start 0%, $color-red-end 100%);

      &--selected{
        &::after{
          position: absolute;
          top: -4px;
          right: 0;
          bottom: 0;
          left: -4px;
          display: block;
          width: 38px;
          height: 38px;
          content: '';
          border: 2px solid $color-red-start;
          border-radius: 50%;
        }
      }
    }

    &__color-3{
      background-image: radial-gradient(circle at 100% 100%,$color-purple-start 0%, $color-red-start 100%);

      &--selected{
        &::after{
          position: absolute;
          top: -4px;
          right: 0;
          bottom: 0;
          left: -4px;
          display: block;
          width: 38px;
          height: 38px;
          content: '';
          border: 2px solid $color-purple-start;
          border-radius: 50%;
        }
      }
    }

    &__color-4{
      background-image: radial-gradient(circle at 100% 100%,$color-blue-start 0%, $color-blue-end 100%);

      &--selected{
        &::after{
          position: absolute;
          top: -4px;
          right: 0;
          bottom: 0;
          left: -4px;
          display: block;
          width: 38px;
          height: 38px;
          content: '';
          border: 2px solid $color-blue-start;
          border-radius: 50%;
        }
      }
    }

    &__color-5{
      background-image: radial-gradient(circle at 100% 100%,$color-green-start 0%, $color-green-end 100%);

      &--selected{
        &::after{
          position: absolute;
          top: -4px;
          right: 0;
          bottom: 0;
          left: -4px;
          display: block;
          width: 38px;
          height: 38px;
          content: '';
          border: 2px solid $color-green-start;
          border-radius: 50%;
        }
      }
    }

    &__color-6{
      background-image: radial-gradient(circle at 100% 100%,$color-yellow-start 0%, $color-yellow-end 100%);

      &--selected{
        &::after{
          position: absolute;
          top: -4px;
          right: 0;
          bottom: 0;
          left: -4px;
          display: block;
          width: 38px;
          height: 38px;
          content: '';
          border: 2px solid $color-yellow-start;
          border-radius: 50%;
        }
      }
    }


  }





}
