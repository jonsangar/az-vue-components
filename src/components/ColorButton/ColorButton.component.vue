<template lang="pug">
.colorButton(
  @click="$emit('click')")
  .colorButton__fill(
    :class="getStyle")
</template>

<script>
export default {
  props: {
    color: {
      type: Number,
      default: 1
    },

    selected: {
      type: Boolean,
      default: false
    }
  },

  computed: {
    getStyle() {
      return [
        `colorButton__fill__color-${this.color}`,
        this.selected ? `colorButton__fill__color-${this.color}--selected` : ''
      ]
    }
  }

}
</script>

<style lang="scss" src="./ColorButton.component.scss" scoped></style>
