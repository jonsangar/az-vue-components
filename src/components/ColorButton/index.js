import ColorButton from './ColorButton.component.vue';

export default ColorButton;
