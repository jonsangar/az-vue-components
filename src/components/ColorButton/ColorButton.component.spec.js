import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import ColorButton from './ColorButton.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('ColorButton.vue', () => {
  const build = () => {
    const wrapper = shallowMount(ColorButton, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente ColorButton con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que emite evento "click" al ser pulsado', async () => {
    const { wrapper } = build();
    // Al pulsar sobre el elemento lanza el evento click
    wrapper.trigger('click');
    // Esperamos para que emits sea manejado
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha emitido el evento click
    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
  });

  it('Comprueba que la propiedad color por defecto no seleccionada', async () => {
    const { wrapper } = build();
    // Comprobamos que se tiene las props por defecto
    expect(wrapper.vm.color).toStrictEqual(1);
    expect(wrapper.vm.selected).toStrictEqual(false);
    // Comprobamos que se ha añadido la clase con color 1
    expect(wrapper.find('.colorButton__fill__color-1').isVisible()).toBe(true);
  });

  it('Comprueba que la propiedad color por defecto seleccionada', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad selected, añade la clase selected
    await wrapper.setProps({
      selected: true,
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha actualizado las props
    expect(wrapper.vm.color).toStrictEqual(1);
    expect(wrapper.vm.selected).toStrictEqual(true);
    // Comprobamos que se ha añadido la clase con color 1
    expect(wrapper.find('.colorButton__fill__color-1').isVisible()).toBe(true);
    // Comprobamos que se ha añadido la clase con color 1 seleccionada
    expect(wrapper.find('.colorButton__fill__color-1--selected').isVisible()).toBe(true);
  });

  it('Comprueba que la propiedad con color 2', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad color, añade la clase colorButton__fill__color-2
    await wrapper.setProps({
      color: 2,
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha actualizado las props
    expect(wrapper.vm.color).toStrictEqual(2);
    // Comprobamos que se ha añadido la clase con color 2
    expect(wrapper.find('.colorButton__fill__color-2').isVisible()).toBe(true);
  });

  it('Comprueba que la propiedad color con valor 2 y seleccionada', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad selected y color, añade la clase selected al color seleccionado
    await wrapper.setProps({
      color: 2,
      selected: true,
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
        // Comprobamos que se ha actualizado las props
    expect(wrapper.vm.color).toStrictEqual(2);
    expect(wrapper.vm.selected).toStrictEqual(true);
    // Comprobamos que se ha añadido la clase con color 2
    expect(wrapper.find('.colorButton__fill__color-2').isVisible()).toBe(true);
    // Comprobamos que se ha añadido la clase con color 2 seleccionada
    expect(wrapper.find('.colorButton__fill__color-2--selected').isVisible()).toBe(true);
  });
});
