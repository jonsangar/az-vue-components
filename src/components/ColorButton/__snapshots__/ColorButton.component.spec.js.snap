// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorButton.vue Comprueba renderizado del componente ColorButton con snapshot 1`] = `
<div class="colorButton">
  <div class="colorButton__fill colorButton__fill__color-1"></div>
</div>
`;
